### PR TITLE
build: add dependabot for `v1` branch, add GitHub Actions

### PR DESCRIPTION
**Pull request by lucacome**
- Target `v1` branch for golang updates until `v2` is GA
- Add check for GitHub Actions

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -18,3 +18,8 @@ updates:
     directory: "/"
     schedule:
       interval: "daily"
+    target-branch: "v1"
```
Comment (shizhMSFT) [inline .github/dependabot.yml:21]: Is it possible to target to both `v1` and `main`?

Comment (lucacome) [reply .github/dependabot.yml:21]: looks like it's not possible unfortunately 😞

Comment (shizhMSFT) [reply .github/dependabot.yml:21]: Will merging this PR to the `v1` branch work?

Comment (lucacome) [reply .github/dependabot.yml:21]: I think it only works from the main branch, but let me do tests in another repo.

Comment (lucacome) [reply .github/dependabot.yml]: @shizhMSFT Turns out you can specify the same config for multiple branches 🎉

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**